src/Tizen.Diagnostics/Tizen.Diagnostics/Diagnostics.cs

```diff
@@ -0,0 +1,285 @@
+/*
+* Copyright (c) 2020 Samsung Electronics Co., Ltd All Rights Reserved
+*
+* Licensed under the Apache License, Version 2.0 (the License);
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using NativeApi = Interop.Diagnostics;
+
+namespace Tizen.Diagnostics
+{
+    /// <summary>
+    /// The Diagnostics class is used for requesting client data and receiving diagnostics report events.
+    /// </summary>
+    /// <since_tizen> 8 </since_tizen>
+    public static class Diagnostics
+    {
+        private static NativeApi.NotificationCallback _notificationCallback;
+        private static event EventHandler<ReportReceivedEventArgs> _reportReceived;
+
+        /// <summary>
+        /// Event to be invoked when new report context arrives.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public static event EventHandler<ReportReceivedEventArgs> ReportReceived
+        {
+            add
+            {
+                if (_reportReceived != null|| value.GetInvocationList().Length > 1)
+                {
+                    throw new InvalidOperationException("Only one handler allowed");
```

Comment:
You must support multiple event handler

Reply:
I have not much experience with dotnet, but I understand that it will take care of memory management of `IntPtr Ctx` and sharing it across all handlers? Native API will create just one instance of it in the callback and it will be destroyed with `Dispose()` after returning from the first handler.

Reply:
When `ReportReceived` was raised, DiagnosticsContext will be created and it will be shared to all event handler. 
if anyone dispose context, all hander can't use it

Anyway, we should support multiple handler,
And app developer be careful using of disposable object.